runtime/lua/vim/glob.lua

```diff
@@ -0,0 +1,84 @@
+local lpeg = vim.lpeg
+
+local M = {}
+
+--- Parses the raw glob into an |lpeg| pattern. LPeg patterns natively support the "this" or "that"
+--- alternative constructions described in the LSP spec that cannot be expressed in a standard Lua pattern.
+---
+---@param pattern string The raw glob pattern
+---@return vim.lpeg.Pattern? pattern An |lpeg| representation of the pattern, or nil if the pattern is invalid.
+function M.parse(pattern)
```

Comment:
If we name this `lpeg()` or `to_lpeg`, that leaves the door open for `to_regex` in the future, and also a `Glob` class if needed. But for now, `glob.lua` is just a namespace for glob-related things.

```suggestion
  function M.parse(pattern)
```